[AOTI] Support libtorch and posix mingw flavor

yushangdi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #165573
* #165560
* __->__ #165557



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

yushangdi committed:
Support libtorch and posix mingw flavor

[ghstack-poisoned]

desertfire reviewed (approved):
Nit: Add AOTI in the title?

yushangdi commented:
ah... I messed up ghstack when updating the PR. New PR is created with changes addressing the comment: https://github.com/pytorch/pytorch/pull/165574
